feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/tindergov/cards/TinderGovCardsViewModel.kt

```diff
@@ -105,17 +108,24 @@ class TinderGovCardsViewModel(
         .map { items -> mapBasketModel(items.values.toList()) }
 
     init {
-        val referendaWithBasket = combineToPair(interactor.observeReferendaAvailableToVote(this), basketFlow)
-        referendaWithBasket.onEach { (referenda, basket) ->
-            val currentReferendaIds = sortedReferendaFlow.value.map { it.id }.toSet()
-            val newReferenda = referenda.filter { it.id !in currentReferendaIds && it.id !in basket }
-
-            sortedReferendaFlow.value += newReferenda // To add new coming referenda to the end of list
-        }.launchIn(this)
+        observeReferendaAndAddToCards()
 
         setupReferendumRetryAction()
 
         loadFirstCards()
+
+        openBasketIfAllReferendaVoted()
+    }
+
+    private fun openBasketIfAllReferendaVoted() {
+        combine(sortedReferendaFlow, basketFlow) { referenda, basket ->
+            if (referenda.isEmpty()) return@combine
+
+            val allReferendaVoted = referenda.all { it.id in basket }
+            if (allReferendaVoted) {
+                router.openTinderGovBasket()
```

Comment:
Should we do it only once? Currently if something changes in `sortedReferendaFlow` (e.g. some referenda state got changed) we will trigger this navigation once more time